VERSIONING.md

```diff
@@ -0,0 +1,198 @@
+- Better to promise a weaker guarantee and stick to it consistently than to be
+  vague about what is/isn’t covered and technically comply, or not comply at
+  all.
+
+## Cosign CLI
+
+The Cosign CLI’s version numbers are of the form `MAJOR.MINOR.PATCH`, but it is
+explicitly not covered by [semantic versioning][] guarantees. Instead, you
+should interpret version numbers as follows:
+
+**Major version increases.** These are "large changes:" you should expect to
+need to rewrite scripts after these. There are no compatibility guarantees about
+the behavior or shape of the CLI. These should be infrequent and correspond to
+big improvements in the user experience.
+
+**Minor version increases.** These are "small changes:" if you’ve been paying
+attention to and fixing deprecation warnings (see "Deprecation"), your scripts
+should continue to work (provided they only depend on explicitly guaranteed
+Cosign behavior; see "What's covered?").
+
+**Patch version increases.** These are "bug fix" releases only.
+
+[semver]: https://semver.org
+
+### What's covered?
+
+Only behavior explicitly listed here is covered (to counter [Hyrum's
+Law][hyrums-law]).
+
+* Any documented behavior (i.e., behavior described in the output of `cosign -h`).
+  If the wording was ambiguous, the behavior may change as long as the
+  corresponding documentation changes to make it less ambiguous.
+* Output to standard output, *when described in the documentation*. Each command
+  should describe its default output format (ideally with examples).
+  * Any scripts should *only* rely on standard output.
+  * Consumers using compliant parsers will continue to be able parse the output of Cosign.
+  * New fields may be added, but fields will not be removed.
+* Additional conventions documented in [CLI.md](CLI.md).
+
+What’s *not* covered?
+
+* Output to standard error is *not* covered and may change at any time.
+* If the Sigstore API changes, the output of Cosign may change.
+* Fixes to security bugs (for instance, something invalid used to pass
+  verification, but now it does not). However, "security features" should not
+  break backwards-compatibility.
+* Any "unstable" commands.
+
+Violations of this are generally considered bugs; we will issue patch releases
+for each affected minor version. In some (rare) cases, an exception may be made
+with the approval of the Cosign maintainers as long as the decision results in
+an update to this versioning policy.
+
+[hyrums-law]: https://www.hyrumslaw.com/
+
+### Unstable commands
+
+It’s important to enable rapid iteration on the development of new Cosign
+features while providing stability to users of more-mature features. As such,
+the Cosign CLI will support the following release levels with corresponding
+guarantees:
+
+* **Experimental**: No guarantees. May change in a backwards-incompatible way or
+  be removed at any time without warning.
+* Generally available (**GA**): follows the guidelines in this document.
```

Comment:
I dunno if this last really matches this "header" of Unstable Commands?

Reply:
changed header to "feature stability"